ci/do_ci.sh

```diff
@@ -58,6 +58,9 @@ function cp_binary_for_image_build() {
 
   # Copy for azp which doesn't preserve permissions, creating a tar archive
   tar czf "${ENVOY_BUILD_DIR}"/envoy_binary.tar.gz -C "${ENVOY_SRCDIR}" build_"$1" build_"$1"_stripped
+
+  # Remove binaries to save space
+  rm -rf "${ENVOY_SRCDIR}"/build_"$1" "${ENVOY_SRCDIR}"/build_"$1"_stripped "${ENVOY_DELIVERY_DIR}"/envoy
```

Comment:
cc @tonya11en I guess this might break your use case when you added `cp_binary_for_outside_access`

Reply:
Hmm yeah not sure why this was added. @tonya11en @LisaLudique can one of you take a look and see if we use this at Lyft somehow?

Reply:
Sorry I'm confused. Why would this break it? We are deleting after the copy which seems OK?

Reply:
Oh I see. You are deleting what we already copied. Yeah I'm sure this is going to break something. Can you maybe guard this with an env var and then just set it for CI?

Reply:
(Or if we decide that we don't want the copy we should just delete cp_binary_for_outside_access() entirely.